app/controllers/products_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ProductsController < ApplicationController
```

Comment:
Examine the indentation here in this file as well, looks like some common copy/paste issues